auctionsystem/client.py

```diff
@@ -98,134 +104,124 @@ def handle_receive(self, data, addr=None):
             pass  # TODO: Something went super wrong, this wasn't a valid command message :O
 
     def rcv_registered(self, req_num, name, ip_addr, port_num):
-        # TODO: Handle UDP message to confirm registration for the client
         self.confirm_acknowledgement(req_num)
-        # TODO: REMOVE THIS, CONTROLLER STUFF SHOULD BE ELSEWHERE
-        desc = ''.join(random.choices(string.ascii_letters + string.digits, k=30))  # Random letters and numbers
-        self.send_offer(self.client_name, self.udp_client.address[0], desc, random.randint(0, 300))
-        print('received registered')
+
+        if self.gui_cb:
+            self.gui_cb(MESSAGE.REGISTERED)
+        else:
+            print('received registered')
 
     def rcv_unregistered(self, req_num, reason):
         # The client could not register with the server
-        msg_args = self.confirm_acknowledgement(req_num)
-        if reason == REASON.ALREADY_REGISTERED.val:
-            # TODO: do nothing
-            print('already registered')
+        self.confirm_acknowledgement(req_num)
+        if self.gui_cb:
+            self.gui_cb(MESSAGE.UNREGISTERED, reason)
         else:
-            # Resend the register message - includes handling for bad IP
-            print('resending register message')
-            self.send_register(name=msg_args[0], ip_addr=msg_args[1], port_num=msg_args[2],
-                               resending=True, req_num_resend=req_num)
+            print('Could not register because {}'.format(reason.str))
 
     def rcv_dereg_conf(self, req_num, name, ip_addr):
+        # Handle UDP message to confirm registration for the client
         self.confirm_acknowledgement(req_num)
-        self.tcp_clients.clear()  # Should close all of the connections
-        self.udp_client.close_socket()
```

Comment:
I think these two lines should not have been removed, prolly got lost during merge conflict resolution

Reply:
Good catch! Added them back in.